sys/arm64/include/atomic.h

```diff
@@ -71,6 +71,12 @@ extern bool lse_supported;
 #define	_ATOMIC_LSE_SUPPORTED	0
 #endif
 
+#ifndef __CHERI_PURE_CAPABILITY__
+#define	_ATOMIC_PTR_CONST	"r"
+#else
+#define	_ATOMIC_PTR_CONST	"C"
+#endif
```

Comment:
I'd still prefer a better name for this, hmm.

Reply:
Maybe `_ATOMIC_PTR_CNSTR` (or if one more letter is ok, `_ATOMIC_PTR_CONSTR`)?

Reply:
Adding that "R" sounds good since it isn't obvious that the CONST means "inlineasm constraint" rather than "const pointer" without looking at the uses of this macro.